week-3/todo-list/script.js

```diff
@@ -1,25 +1,128 @@
+        listing.style.textDecoration = "none";
+      } else {
+        listing.style.textDecoration = "line-through";
+      }
+    });
+
+
+    rem.addEventListener("click", () => {
+      listing.remove();
+    });
+  });
+ }
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
 let todos = [
   { task: "Wash the dishes", completed: false },
   { task: "Do the shopping", completed: false },
 ];
 
+
 populateTodoList(todos);
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
-  event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
-}
+  const inputTodo = document.getElementById("textboxinput");
+  const buttonTodo = document.querySelector("button");
+  buttonTodo.addEventListener("click", (event) => {
+    event.preventDefault();
+    const newList = document.createElement("li");
+    let list = document.getElementById("todo-list");
+    const tex = document.createTextNode(inputTodo.value);
+    inputTodo.value = "";
+    const rem = document.createElement("span");
+    const add = document.createElement("span");
+    const dateCale = document.createElement("span");
+    const dateInput = document.createElement("input");
+    dateInput.type = "date";
+    dateInput.classList.add("inputing");
+    dateInput.style.display = "none";
+    rem.textContent = "🗑";
+    add.textContent = "✅";
+    dateCale.textContent = "📅";
+    newList.appendChild(rem);
+    newList.appendChild(add);
+    newList.appendChild(dateCale);
+    newList.appendChild(dateInput);
+    newList.append(tex);
+    list.appendChild(newList);
+
+
+    let clickCount = 0;
 
-// Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+
+    dateCale.addEventListener("click", () => {
+      if (dateInput.style.display === "none") {
+        dateInput.style.display = "block";
+      } else {
+        dateInput.style.display = "none";
+      }
+    });
+    add.addEventListener("click", () => {
+      clickCount++;
+      let isEvenClick = clickCount % 2 === 0;
+
+
+      if (isEvenClick) {
+        newList.style.textDecoration = "none";
+      } else {
+        newList.style.textDecoration = "line-through";
+      }
+    });
+
+
+    rem.addEventListener("click", () => {
+      newList.remove();
+    });
+  });
+
+
+  }
+addNewTodo();
 function deleteAllCompletedTodos() {
-  // Write your code here...
+  const del = document.getElementById("remove-all-completed");
```

Comment:
this function isnt implemented correctly, when clicked, it removes all todos on the board. You should rewrite this function to filter out todos that has their `completed` status as `true`